sherpa/fit.py

```diff
@@ -977,20 +961,26 @@ def guess(self, **kwargs):
         """
         self.model.guess(*self.data.to_guess(), **kwargs)
 
+    # QUS: should this have an @evaluates_model decorator?
+    #
     def _calc_stat(self):
-        dep, staterror, syserror = self.data.to_fit(self.stat.calc_staterror)
-        model = self.data.eval_model_to_fit(self.model)
-        extra_args = self._iterfit.get_extra_args(dep)
-        return self.stat.calc_stat(dep, model, staterror, syserror=syserror,
-                                   extra_args=extra_args)
+        """Calculate the current statistic value.
+
+        Returns
+        -------
+        statval, fvec : number, array of numbers
+            The overall statistic value and the "per-bin" value.
+        """
+
+        # TODO: is there anything missing here that
+        #       self._iterfit.get_extra_args calculates?
+        return self.stat.calc_stat(self.data, self.model)
```

Comment:
This is not necessarily a show stopper, but wouldn't it be more generic to pass the entire object, rather than just `self.data` and `self.model`? This way more specific statistics' `calc_stat` methods could subset more specific fit objects and use more information that would not be available otherwise, providing more flexibility.